app/src/main/java/cz/muni/fi/a2p06/stolencardatabase/MainActivity.java

```diff
@@ -20,9 +30,18 @@
     @BindView(R.id.fragment_container)
     FrameLayout mFragmentContainer;
 
+    FirebaseAuth mAuth;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mAuth = FirebaseAuth.getInstance();
+        if (mAuth.getCurrentUser() == null) {
+            Intent loginIntent = new Intent(MainActivity.this, SignInActivity.class);
+            startActivity(loginIntent);
+            finish();
```

Comment:
I'm not sure if this is called after startActivity() method. Try to click the back button after SignInActivity is launched.
Maybe you can place this if statement inside onStart() method.